From 874b7786fd9032fab802ec4007a5e3c010fe24a5 Mon Sep 17 00:00:00 2001
From: skufly <skufly@163.com>
Date: Wed, 8 Jan 2020 15:04:06 +0800
Subject: [PATCH] chore: enable bitcode

---
 meson.build | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 2bcd0b8..db04f4e 100644
--- a/meson.build
+++ b/meson.build
@@ -27,7 +27,10 @@ project('dav1d', ['c'],
     default_options: ['c_std=c99',
                       'warning_level=2',
                       'buildtype=release',
-                      'b_ndebug=if-release'],
+                      'b_ndebug=if-release',
+                      'b_bitcode=true',
+                      'b_asneeded=false',
+                      ],
     meson_version: '>= 0.47.0')
 
 dav1d_soname_version       = '3.1.0'
-- 
2.19.1

